### ISSUE TITLE
Write ENDF format

### ISSUE THREAD

**Issue by dddijulio** [mctools/ncrystal#129]
Add feature to output ENDF format.

**Comment by marquezj** [comment]
From the discussion in the school, it might be useful to use [FUDGE](https://github.com/LLNL/fudge) to generate data. FUDGE provides an API to write nuclear data in the Generalized Nuclear Data format, and tools to export to ENDF-6. This is the [definition of the format](https://www.oecd-nea.org/jcms/pl_39689/specifications-for-the-generalised-nuclear-database-structure-gnds), although it was recently updated to version 2.0.

**Comment by gpanobre** [comment]
There were significant changes from GNDS-1.9 to GNDS-2.0. I strongly encourage to go straight to the 2.0 version. The documentation and definitions are in a NEA document that is currently in press. As soon as it's out I can post here. In the meantime, if you want to have a sneak peek, please contact me and I can share it privately.

**Comment by tkittel** [comment]
Just for my information, how large is the number of people using ENDF6 versus GNDS currently?

**Comment by xxcai1** [comment]
I'm looking at the documentation for v1.9. In addition to generating ENDF files, it has a MonteCarlo_cdf style type. Anyone is using it directly in a transport code for fast neutrons?

**Comment by marquezj** [comment]
@tkittel : as far as I know: ENDF-6 is the current established format, and GNDS is being proposed. GNDS has the advantage of including new physics (ENDF-6 is quite limited). 

If we want to go directly to ENDF-6 now, an alternative package is [ENDFtk](https://github.com/njoy/ENDFtk). [Pyne](https://pyne.io/) and [openmc.data](https://docs.openmc.org/en/stable/pythonapi/generated/openmc.data.ThermalScattering.html) read but do not write ENDF-6 files.

**Comment by tkittel** [comment]
Thanks @marquezj that is what I thought. So I imagine if we are providing the "ncmat2x" feature as a short-term service to an existing community not able to directly use NCrystal, it makes most sense if x==ENDF6. Of course, if it is easy to support GNDS as well, I don't have any problem with also supporting that.

In any case, I'll be happy to help with the "get data out of ncrystal" part, and will be 100% relying on volunteers for the "to x" part :-)

**Comment by marquezj** [comment]
Yes, we have all the information there already.  Most of it explicitly, and then `NCrystal.vdos.analyseVDOS()` gives the Teff. We also have the algorithm from NJOY+NCrystal to decide what we put in the ENDF-6 file (how to separate the components, how to fit the constrains of the format). So the only missing piece is a reliable converter.

**Comment by tkittel** [comment]
Slightly off-topic, but @marquezj there is even a method directly on the dyninfo object, so you don't have to worry about transferring the correct parameters to `NCrystal.vdos.analyseVDOS()`:

![image](https://github.com/mctools/ncrystal/assets/20966041/eb6f8d72-a7dc-462d-b4c1-402ae1eb3955)

**Comment by marquezj** [comment]
Great, that is handy!

**Comment by gpanobre** [comment]
Hi Thomas and all,

Just complementing a bit Ignacio's comment: Yes, ENDF-6 is the established format and if you want to reach a broader range of users you can't avoid generating ENDF-6 formatted files. Just keep in mind that, for the uninitiated, the ENDF-6 format can be quite daunting. So you will have to decide what makes more sense to you, understanding the ENDF-6 format so you can directly output in that format, or starting with a more modern format, GNDS, and use things like FUDGE to do one-to-one translations to ENDF. ENDF-6 is still supported and will still be fully supported for quite a while, but the next release of ENDF/B, version VIII.1, will be officially released in both ENDF-6 and GNDS, and more and more evaluation and processing codes are building now interfaces to operate directly in GNDS. For more detailed questions feel free to reach out to Caleb Mattoon (LLNL) and Dave Brown (BNL)

**Comment by gpanobre** [comment]
Xiao Xiao, I know the Livermore folks use MCGIDI and GIDI+ as event generators and they "speak" ENDF-6 and GNDS. You should probably contact Caleb Mattoon and Bret Beck (LLNL) on that.

**Comment by tkittel** [comment]
Related to #183

**Comment by marquezj** [comment]
@gpanobre NCrystal 4.2.0 is out, and it supports exporting ENDF-6 files.

[Release notes](https://github.com/mctools/ncrystal/releases/tag/v4.2.0)

[ncmat2endf documentaton](https://github.com/mctools/ncrystal/wiki/ncmat2endf)

**Comment by tkittel** [comment]
Indeed, so closing here :-)
